Bindings/Java/Matlab/examples/Moco/example2DWalking/example2DWalkingStepAsymmetry.m

```diff
@@ -0,0 +1,452 @@
+%                                                                              %
+% Author(s): Russell T. Johnson                                                %
+%            University of Southern California, rtjohnso@usc.edu               % 
+%                                                                              %
+% Licensed under the Apache License, Version 2.0 (the "License"); you may      %
+% not use this file except in compliance with the License. You may obtain a    %
+% copy of the License at http://www.apache.org/licenses/LICENSE-2.0            %
+%                                                                              %
+% Unless required by applicable law or agreed to in writing, software          %
+% distributed under the License is distributed on an "AS IS" BASIS,            %
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.     %
+% See the License for the specific language governing permissions and          %
+% limitations under the License.                                               %
+% ---------------------------------------------------------------------------- %
+
+% Simulate asymmetric gait using MocoStepTimeAsymmetryGoal and 
+% MocoStepLengthAsymmetryGoal. This is an extension of the example2DWalking 
+% MATLAB example (see example2DWalking.m for details about the model and data 
+% used).
+function example2DWalkingStepAsymmetry()
+
+% Simulate asymmetric step times using MocoStepTimeAsymmetryGoal.
+stepTimeAsymmetry();
+
+% Simulate asymmetric step lengths using MocoStepLengthAsymmetryGoal.
+stepLengthAsymmetry();
+
+end
+
+% Step Time Asymmetry
+% -------------------
+% Set up a predictive optimization problem where the goal is to minimize an 
+% effort cost (cubed controls) and hit a target step time asymmetry. Unlike 
+% example2DWalking, this problem requires simulating a full gait cycle. 
+% Additionally, endpoint constraints enforce periodicity of the coordinate values 
+% (except for pelvis tx) and speeds, coordinate actuator controls, and muscle 
+% activations.
+%
+% Step time is defined as the time between consecutive foot strikes. Step Time 
+% Asymmetry (STA) is a ratio and is calculated as follows:
+%  - Right Step Time (RST) = Time from left foot-strike to right foot-strike
+%  - Left Step Time (LST)  = Time from right foot-strike to left foot-strike
+%  - STA = (RST - LST) / (RST + LST)
+%
+% Asymmetry values range from -1.0 to 1.0. For example, 0.20 is 20% positive 
+% step time asymmetry with greater right step time than left step time.
+%
+% The step time goal works by "counting" the number of nodes that each foot is in 
+% contact with the ground (with respect to a specified contact force threshold). 
+% Since, in walking, there are double support phases where both feet are on the 
+% ground, the goal also detects which foot is in front and assigns the step time 
+% to the leading foot. Altogether, it calculates the time between consecutive 
```

Comment:
I think we should changes this to "Altogether, it _estimates_ the time between consecutive heel strikes in order to _infer_ left and right step times"